internal/controller/apiextensions/composite/composition_functions.go

```diff
@@ -346,6 +346,7 @@ func (c *FunctionComposer) Compose(ctx context.Context, xr *composite.Unstructur
 		// Used to store the response of the function at the previous iteration.
 		var rsp *v1beta1.RunFunctionResponse
 
+	extraResourcesLoop:
```

Comment:
😭 I've always thought this named loop feature was some of the ugliest syntax in all of Go.

That's subjective though, so not a blocker. On the other hand the broader pipeline loop feels like it's become very very long. I think it's getting a bit hard to reason about. I might try see if it's possible to restructure today, maybe break some of it out into distinct functions.

Reply:
Yeah, @negz, mixed feelings here too, it has a vintage look that I kind of like, but I get your point 😂 I just didn't want to refactor it the day before cutting the release, but given that tomorrow I'll start the code freeze I can refactor it before merging.